sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py

```diff
@@ -140,14 +141,16 @@ def set_role_definition(self, role_scope, permissions, **kwargs):
         ]
 
         properties = self._client.role_definitions.models.RoleDefinitionProperties(
-            role_name=role_definition_name, permissions=permissions, **kwargs
+            role_name=kwargs.pop("role_name", None),
+            description=kwargs.pop("description", None),
+            permissions=permissions
         )
         parameters = self._client.role_definitions.models.RoleDefinitionCreateParameters(properties=properties)
 
         definition = self._client.role_definitions.create_or_update(
             vault_base_url=self._vault_url,
             scope=role_scope,
-            role_definition_name=role_definition_name,
+            role_definition_name=kwargs.pop("role_definition_name", None) or uuid4(),
```

Comment:
The generated client expects a string. It may work anyway with a uuid.UUID due to some implementation detail, but it will be safer to pass a string.